src/Tasks.UnitTests/XmlPoke_Tests.cs

```diff
@@ -24,16 +24,16 @@ sealed public class XmlPoke_Tests
     {
         private const string XmlNamespaceUsedByTests = "http://nsurl";
 
-        private string _xmlFileWithNs = $@"<?xml version='1.0' encoding='utf-8'?>
+        private const string _xmlFileWithNs = @"<?xml version='1.0' encoding='utf-8'?>
         
-<class AccessModifier='public' Name='test' xmlns:s='{XmlNamespaceUsedByTests}'>
+<class AccessModifier='public' Name='test' xmlns:s='" + XmlNamespaceUsedByTests + @"'>
```

Comment:
Not sure I like this.

Reply:
The resulting string is the same, except that it becomes a `const`. Why do you have a preference for an instance field?

Reply:
I consider it cleaner to use $ and {} in allocating strings over just concatenating them, and it should theoretically mean fewer allocations (not that that matters here). I do like the const part, though, which is why I don't feel at all strongly.

Reply:
That would be true for some versions of the compiler and some uses of string interpolation.

In this case, there is no run-time concatenation of strings and no allocation, because it became an interned constant string.

Look [here](https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCsDetE8QBwCcBLANwEMAXAU0QGMB7AOwGcLkIAGRAZQkQF5EAIgAeQgNzZchUpRoMWbEJzRNWAVWZFGAc14DhZIYgDUPCJJi58xctXYA2RASpkAJkwA2AT3ZdUahSa2joAkozUBHj0npRETAYAJEJkGLwAvhKw6UA).

Reply:
Since the compiler [does the constant expansion in the previous form](https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgYGYAE9MGFUDeqmJmADgE5gBuAhsAKaYDGA9gHYDOw2CADJgAaAWwgA5WsIacytZgwCqnBgBMAQgE8AKtOCdMAXkwAiABbBgZEAHprXAK4UIxgNzFSlGvSYx+mAPoAHqIAYpAMAOpgwKZi+kYAJDAwxgA8APzBEJjUDBScYBwGAOQIAHR8xZgM7GwqYOwA5iX2wABmALQAHMXpAHzupKSoqTgAgszynJwAsqz1bWB5JRhVElIljNxVWVwgnCUEIuKS0rLySqqaOtycAL7FAygkqZwgdFS0AEYQTFoaZAYJQAysAqE01qcSrRHqlrG8PmBvr8ni8EbRPj8/gCgcVQeDGpCNsUvrD4e8MUisajMK8KZjfph/oCQWCGoTMOtccwyeiGQwaakpDF5pgJlNZvMwItlsUMLwAGxE3EAcQYwAAarQIFVrE84fA+q5UHcgA==), I also weigh the "better readability" factor higher than the "formalized `const`" factor here. Especially in a test assembly.

Reply:
"better readability" is in the eye of the reader. The fact that it wasn't even a `static readonly` immediately caught my eye.

The difference between `const` and an instance field is more than a mere formality. A `const` means that every constant value will have one and only one existence in the application domain. An instance field is instantiated per instantiation of the class. Every instance of the class will have its own copy of that string. For XUnit.NET, that means an instance per instance `Fact` method (`Fact` methods don't need to be instance methods, they can be static methods).